x-pack/plugins/apm/server/routes/mobile/get_mobile_main_statistics_by_field.ts

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  termQuery,
+  kqlQuery,
+  rangeQuery,
+} from '@kbn/observability-plugin/server';
+import { ProcessorEvent } from '@kbn/observability-plugin/common';
+import {
+  APP_LAUNCH_TIME,
+  SERVICE_NAME,
+  SESSION_ID,
+  TRANSACTION_DURATION,
+  ERROR_TYPE,
+  AGENT_NAME,
+} from '../../../common/es_fields/apm';
+import { environmentQuery } from '../../../common/utils/environment_query';
+import { APMEventClient } from '../../lib/helpers/create_es_client/create_apm_event_client';
+import { getLatencyValue } from '../../lib/helpers/latency_aggregation_type';
+import { LatencyAggregationType } from '../../../common/latency_aggregation_types';
+import { calculateThroughputWithRange } from '../../lib/helpers/calculate_throughput';
+
+interface Props {
+  kuery: string;
+  apmEventClient: APMEventClient;
+  serviceName: string;
+  environment: string;
+  start: number;
+  end: number;
+  field: string;
+}
+
+export interface MobileMainStatisticsResponse {
+  mainStatistics: Array<{
+    name: string | number;
+    latency: number | null;
+    throughput: number;
+    crashRate: number;
+    appLaunchTime: number | null;
+  }>;
+  agentName?: string;
+}
+
+export async function getMobileMainStatisticsByField({
+  kuery,
+  apmEventClient,
+  serviceName,
+  environment,
+  start,
+  end,
+  field,
+}: Props) {
+  const response = await apmEventClient.search(
+    `get_mobile_main_statistics_by_${field}`,
+    {
+      apm: {
+        events: [
+          ProcessorEvent.transaction,
+          ProcessorEvent.error,
+          ProcessorEvent.metric,
+        ],
+      },
+      body: {
+        track_total_hits: false,
+        size: 1,
```

Comment:
I think we should avoid getting documents and aggs in a single request, can we break them up?

Reply:
I needed to retrieve the agent.name, I will see if I can get it from somewhere (maybe we're already getting it in a higher level) or create another request

Reply:
we should already have an API endpoint that returns the agent name for a service.